test/test_utils/security.py

```diff
@@ -187,3 +262,280 @@ def get_ecr_scan_allowlist_path(image_uri):
+    apt_upgrade_list_filename = f"apt-upgrade-list-{test_utils.get_processor_from_image_uri(image)}.txt"
+    s3_filename_for_apt_upgrade_list = s3_filename_for_allowlist.replace("allowlist.json", apt_upgrade_list_filename)
+    original_filepath_for_apt_upgrade_list = "/".join(
+        original_filepath_for_allowlist.split("/")[:-1] + [apt_upgrade_list_filename]
+    )
+    new_package_list = fixable_list if isinstance(fixable_list, list) else list(fixable_list.keys())
+    create_and_save_package_list_to_s3(
+        original_filepath_for_apt_upgrade_list,
+        new_package_list,
+        s3_filename_for_apt_upgrade_list,
+        s3_bucket_for_storage,
+    )
+    edited_files.append(
+        {"s3_filename": s3_filename_for_apt_upgrade_list, "github_filepath": original_filepath_for_apt_upgrade_list}
+    )
+    newly_found_non_fixable_list = {}
+    if newly_found_non_fixable_vulnerabilites:
+        newly_found_non_fixable_list = newly_found_non_fixable_vulnerabilites.vulnerability_list
+    message_body = {
+        "edited_files": edited_files,
+        "fixable_vulnerabilities": fixable_list,
+        "non_fixable_vulnerabilities": newly_found_non_fixable_list,
+    }
+    _invoke_lambda(function_name="trshanta-ECR-AS", payload_dict=message_body)
+    return_dict = copy.deepcopy(message_body)
+    return_dict["s3_filename_for_allowlist"] = s3_filename_for_allowlist
+    return_dict["s3_filename_for_current_image_ecr_scan_list"] = s3_filename_for_current_image_ecr_scan_list
+    return return_dict
+
+
+def process_failure_routine_summary_and_store_data_in_s3(failure_routine_summary, s3_bucket_name):
+    """
+    This method is especially constructed to process the failure routine summary that is generated as a result of 
+    calling conduct_failure_routine. It extracts lists and calls the save lists function to store them in the s3
+    bucket.
+
+    :param failure_routine_summary: dict, dictionary returned as an outcome of conduct_failure_routine method
+    :param s3_bucket_name: string, name of the s3 bucket
+    :return s3_filename_for_fixable_list: string, filename in the s3 bucket for the fixable vulnerabilities
+    :return s3_filename_for_non_fixable_list: string, filename in the s3 bucket for the non-fixable vulnerabilities
+    """
+    s3_filename_for_allowlist = failure_routine_summary["s3_filename_for_allowlist"]
+    s3_filename_for_fixable_list = s3_filename_for_allowlist.replace(
+        "allowlist.json", "fixable-vulnerability-list.json"
+    )
+    s3_filename_for_non_fixable_list = s3_filename_for_allowlist.replace(
+        "allowlist.json", "non-fixable-vulnerability-list.json"
+    )
+    save_details = []
+    save_details.append((s3_filename_for_fixable_list, failure_routine_summary["fixable_vulnerabilities"]))
+    save_details.append((s3_filename_for_non_fixable_list, failure_routine_summary["non_fixable_vulnerabilities"]))
+    _save_lists_in_s3(save_details, s3_bucket_name)
```

Comment:
review note: If you are pushing these files to S3 anyway, an alternative approach is to just put an Event Trigger on the Lambda Function, with the event set to be the addition of a new file to the S3 bucket at some expected folder (or just anywhere in the bucket).

However, I don't know the time bounds on S3's Strong Consistency, now, so I think this approach is good as is.

Reply:
I think it could have been done, however, as per the implementation, the PR would be created only when the code is running in canary context. In other words, the contents will be uploaded in the s3 file everytime, however, the lambda should not be invoked everytime. So, I even I think that this approach is good for the time being.